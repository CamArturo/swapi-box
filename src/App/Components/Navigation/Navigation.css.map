{
"version": 3,
"mappings": "AACA,GAAI;EACF,QAAQ,EAAE,QAAQ;EAClB,MAAM,EAAE,CAAC;EACT,MAAM,EAAE,IAAI;EACZ,OAAO,EAAE,IAAI;EACb,eAAe,EAAE,MAAM;EACvB,KAAK,EAAE,IAAI;EACX,kBAAkB,EAAE,iBAAiB;EACrC,eAAe,EAAE,iBAAiB;EAClC,aAAa,EAAE,iBAAiB;EAChC,UAAU,EAAE,iBAAiB;EAC7B,MAAG;IACD,MAAM,EAAE,CAAC;IACT,OAAO,EAAE,CAAC;IACV,eAAe,EAAE,IAAI;IACrB,SAAG;MACD,MAAM,EAAE,MAAM;MACd,KAAK,EAAE,IAAI;MACX,WAAE;QACA,eAAe,EAAE,IAAI;QACrB,KAAK,EAAE,OAAO;QACd,WAAW,EAAE,GAAG;QAChB,SAAS,EAAE,IAAI;;AAMvB,iCAAiC;AAKjC,8BAA8B;AAK9B,4BAA4B;AAK5B,iCAAiC;AAKjC,2BAA2B",
"sources": ["Navigation.scss"],
"names": [],
"file": "Navigation.css"
}