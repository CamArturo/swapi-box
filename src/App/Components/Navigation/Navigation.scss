
nav {
  position: absolute;
  bottom: 0;
  height: 4rem;
  display: flex;
  justify-content: center;
  width: 100%;
  -webkit-transition: all 0.3s ease-out;
  -moz-transition: all 0.3s ease-out;
  -o-transition: all 0.3s ease-out;
  transition: all 0.3s ease-out;
  ul {
    margin: 0;
    padding: 0;
    list-style-type: none;
    li {
      margin: 0 1rem;
      float: left;
      a {
        text-decoration: none;
        //color: #1919FF;
        color: #2a2a57;
        font-weight: 600;
        font-size: 2rem;
      }
    }
  }
}

/* Large Devices, Wide Screens */
@media only screen and (max-width: 1200px) {

}

/* Medium Devices, Desktops */
@media only screen and (max-width: 992px) {

}

/* Small Devices, Tablets */
@media only screen and (max-width: 768px) {

}

/* Extra Small Devices, Phones */
@media only screen and (max-width: 480px) {

}

/* Custom, iPhone Retina */
@media only screen and (max-width: 320px) {

}