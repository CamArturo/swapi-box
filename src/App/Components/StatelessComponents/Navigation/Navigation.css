nav {
  position: fixed;
  overflow: hidden;
  margin-bottom: 1rem;
  bottom: 0;
  height: 4rem;
  display: flex;
  justify-content: center;
  width: 100%;
  -webkit-transition: all 0.3s ease-out;
  -moz-transition: all 0.3s ease-out;
  -o-transition: all 0.3s ease-out;
  transition: all 0.3s ease-out; }
  nav button {
    text-decoration: none;
    color: rgba(255, 255, 255, 0.8);
    background: #2a2a57;
    padding: 7px 1rem;
    margin: 1rem;
    text-align: center;
    font-weight: normal;
    text-transform: uppercase;
    transition: all 0.2s ease-in-out;
    cursor: pointer; }
    nav button:hover {
      color: white;
      box-shadow: 0 5px 15px rgba(42, 42, 87, 0.9); }

/* Large Devices, Wide Screens */
/* Medium Devices, Desktops */
/* Small Devices, Tablets */
/* Extra Small Devices, Phones */
/* Custom, iPhone Retina */

/*# sourceMappingURL=Navigation.css.map */
