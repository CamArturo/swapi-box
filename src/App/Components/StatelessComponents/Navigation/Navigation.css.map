{
"version": 3,
"mappings": "AACA,GAAI;EACF,QAAQ,EAAE,KAAK;EACf,QAAQ,EAAE,MAAM;EAChB,aAAa,EAAE,IAAI;EACnB,MAAM,EAAE,CAAC;EACT,MAAM,EAAE,IAAI;EACZ,OAAO,EAAE,IAAI;EACb,eAAe,EAAE,MAAM;EACvB,KAAK,EAAE,IAAI;EACX,kBAAkB,EAAE,iBAAiB;EACrC,eAAe,EAAE,iBAAiB;EAClC,aAAa,EAAE,iBAAiB;EAChC,UAAU,EAAE,iBAAiB;EAC7B,UAAO;IACL,eAAe,EAAE,IAAI;IACrB,KAAK,EAAE,wBAAwB;IAC/B,UAAU,EAAE,OAAO;IACnB,OAAO,EAAE,QAAQ;IACjB,MAAM,EAAE,IAAI;IACZ,UAAU,EAAE,MAAM;IAClB,WAAW,EAAE,MAAM;IACnB,cAAc,EAAE,SAAS;IACzB,UAAU,EAAE,oBAAoB;IAChC,MAAM,EAAE,OAAO;IACf,gBAAQ;MACN,KAAK,EAAE,KAAsB;MAC7B,UAAU,EAAE,gCAA+B;;AAKjD,iCAAiC;AAKjC,8BAA8B;AAK9B,4BAA4B;AAK5B,iCAAiC;AAKjC,2BAA2B",
"sources": ["Navigation.scss"],
"names": [],
"file": "Navigation.css"
}