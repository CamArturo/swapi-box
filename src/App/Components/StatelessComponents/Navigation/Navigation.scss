
nav {
  position: fixed;
  overflow: hidden;
  bottom: 0;
  height: 5rem;
  display: flex;
  justify-content: center;
  width: 100%;
  background: rgba(44, 44, 44, 0.85);
  -webkit-transition: all 0.3s ease-out;
  -moz-transition: all 0.3s ease-out;
  -o-transition: all 0.3s ease-out;
  transition: all 0.3s ease-out;
  button {
    text-decoration: none;
    color: rgba(255, 255, 255, 0.8);
    background: #2a2a57;
    padding: 7px 1rem;
    margin: 1rem;
    text-align: center;
    font-weight: normal;
    text-transform: uppercase;
    transition: all 0.2s ease-in-out;
    cursor: pointer;
    width: 120px;
    &:hover {
      color: rgba(255, 255, 255, 1);
      //background-color: #414141;
      box-shadow: 0 0 6px 3px rgba(255, 209, 0, 1);
    }
  }
}

.active {
  background-color: #ffd100;
  color: #2a2a57;
}

/* Large Devices, Wide Screens */
@media only screen and (max-width: 1200px) {

}

/* Medium Devices, Desktops */
@media only screen and (max-width: 992px) {

}

/* Small Devices, Tablets */
@media only screen and (max-width: 768px) {

}

@media only screen and (max-width: 630px) {
  nav {
    height: auto;
    flex-wrap: wrap;
    button {
      padding: 10px;
      margin: 10px;
      width: 100px;
      //width: auto;
    }
  }
}

/* Extra Small Devices, Phones */
@media only screen and (max-width: 480px) {
  nav {
    button {
      font-size: 12px;
    }
  }
}

/* Custom, iPhone Retina */
@media only screen and (max-width: 320px) {

}