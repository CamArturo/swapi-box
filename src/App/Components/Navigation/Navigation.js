import React from 'react';
import PropTypes from 'prop-types';
import './Navigation.css';

const Navigation = props => {
  return (
    <nav id="menu" className="fadein">
      <ul>
        <li><a href="#">People</a></li>
        <li><a href="#">Planets</a></li>
        <li><a href="#">Vehicles</a></li>
        <li><a href="#">Favorites</a></li>
      </ul>
    </nav>
  );
};

Navigation.propTypes = {
  
};

export default Navigation;
