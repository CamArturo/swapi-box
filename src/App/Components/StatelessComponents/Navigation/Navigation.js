import React from 'react';
import './Navigation.css';
import PropTypes from 'prop-types';

const Navigation = props => {
  return (
    <nav id="menu" className="fadein">
      <button onClick={() => {
        props.selectCategory('people');
      }} >People</button>
      <button onClick={() => {
        props.selectCategory('planets');
      }}>Planets</button>
      <button onClick={() => {
        props.selectCategory('vehicles');
      }}>Vehicles</button>
      <button >Favorites</button>
    </nav>
  );
};

Navigation.propTypes = {
  selectCategory: PropTypes.func.isRequired
  // updateFavorites: PropTypes.func.isRequired
};

export default Navigation;
