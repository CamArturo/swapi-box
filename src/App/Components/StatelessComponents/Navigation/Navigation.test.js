import React from 'react';
import {shallow} from 'enzyme';
import Navigation from "./Navigation";


describe('Navigation test', () => {
  let wrapper;
  const selectCategory = jest.fn();
  const displayFavorites = jest.fn();
  const favorites = 3;
  const currentCategory = 'planet';

  const vehicles = [
    {
      name: "Sand Crawler",
      model: "Digger Crawler",
      vehicleClass: "wheeled",
      passengers: "30",
      favorite: false
    },
    {
      name: "T-16 skyhopper",
      model: "T-16 skyhopper",
      vehicleClass: "repulsorcraft",
      passengers: "1",
      favorite: false
    }
  ];

  beforeEach(() => {

    wrapper = shallow(<Navigation selectCategory={selectCategory}
                                  displayFavorites={displayFavorites}
                                  favorites={favorites}
                                  currentCategory={currentCategory}
    />);
  });
  it('should match the snapshot', () => {

    expect(wrapper).toMatchSnapshot();
  });

  it('should call the selectCategory prop when clicked', () => {

    wrapper.find('.selectPeople').simulate('click');

    expect(selectCategory).toHaveBeenCalled();
  });

  it('should selectPlanets button when clicked should call the selectCategory prop when clicked', () => {

    wrapper.find('.selectPlanets').simulate('click');

    expect(selectCategory).toHaveBeenCalled();
  });

  it('should selectVehicles button when clicked should call the selectCategory prop when clicked', () => {

    wrapper.find('.selectVehicles').simulate('click');

    expect(selectCategory).toHaveBeenCalled();
  });

  it('should selectFavorites button should call the selectCategory prop when clicked', () => {

    wrapper.find('.selectFavorites').simulate('click');

    expect(displayFavorites).toHaveBeenCalled();
  });
});
