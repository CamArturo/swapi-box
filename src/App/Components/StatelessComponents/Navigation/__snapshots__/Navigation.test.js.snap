// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation test should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navigation
    currentCategory="planet"
    displayFavorites={[Function]}
    favorites={3}
    selectCategory={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="selectPeople"
          onClick={[Function]}
>
          People
</button>,
        <button
          className="selectPlanets"
          onClick={[Function]}
>
          Planets
</button>,
        <button
          className="selectVehicles"
          onClick={[Function]}
>
          Vehicles
</button>,
        <button
          className="selectFavorites"
          onClick={[Function]}
>
          Favorites 
          3
</button>,
      ],
      "className": "fadein",
      "id": "menu",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "People",
          "className": "selectPeople",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "People",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Planets",
          "className": "selectPlanets",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Planets",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Vehicles",
          "className": "selectVehicles",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Vehicles",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Favorites ",
            3,
          ],
          "className": "selectFavorites",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          "Favorites ",
          3,
        ],
        "type": "button",
      },
    ],
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="selectPeople"
            onClick={[Function]}
>
            People
</button>,
          <button
            className="selectPlanets"
            onClick={[Function]}
>
            Planets
</button>,
          <button
            className="selectVehicles"
            onClick={[Function]}
>
            Vehicles
</button>,
          <button
            className="selectFavorites"
            onClick={[Function]}
>
            Favorites 
            3
</button>,
        ],
        "className": "fadein",
        "id": "menu",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "People",
            "className": "selectPeople",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "People",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Planets",
            "className": "selectPlanets",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Planets",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Vehicles",
            "className": "selectVehicles",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Vehicles",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Favorites ",
              3,
            ],
            "className": "selectFavorites",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            "Favorites ",
            3,
          ],
          "type": "button",
        },
      ],
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
